cli/src/ui.rs

```diff
@@ -130,7 +131,35 @@ impl UiOutput {
                 }
             }
             UiOutput::BuiltinPaged { pager } => {
-                pager.finalize();
+                if let Err(minus_error) = pager.finalize() {
+                    writeln!(
+                        ui.error_with_heading("Error"),
+                        "Built-in pager failed to start: {minus_error}",
+                    )
+                    .ok();
+                    writeln!(ui.error_no_heading(), "The output of this command is lost.").ok();
+                    if let MinusError::Setup(minus::error::SetupError::InvalidTerminal) =
+                        minus_error
```

Comment:
nit: `matches!(..)` instead of `let`?

And maybe we can add `cfg!(windows)` because the added hint is very Windows specific.

Reply:
Well, the first three workarounds still would work if this error shows up not on Windows. I could write two different hints for Windows and non-Windows, but I'd rather do that if we hear about somebody seeing this error in a different situation.

Reply:
I prefer a shorter hint in general, and I would add "Hint: Try `jj --no-pager`" or something on non-Windows if we need to show something.

The builtin pager isn't default on non-Windows, and it's less likely that a terminal emulator has compatibility issue on non-Windows. (If it had, a better suggestion would be to try `TERM=xterm`.)

---

Unrelated: maybe better to not shout "POSSIBLE WORKAROUNDS:"? I think "Possible workarounds:" is our style.

Reply:
I simplified the hint a bit and set up a shorter hint for non-Windows. I'll probably merge this shortly, we can keep editing the hint after that as well.